content/master/concepts/compositions.md

```diff
@@ -0,0 +1,993 @@
+
+```yaml {label="envpatch",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  environment:
+  # Removed for Brevity
+  resources:
+  # Removed for Brevity
+    - name: vpc
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: VPC
+        spec:
+          forProvider:
+            cidrBlock: 172.16.0.0/16
+      patches:
+        - type: ToEnvironmentFieldPath
+          fromFieldPath: status.atProvider.id
+          toFieldPath: vpcId
+        - type: FromEnvironmentFieldPath
+          fromFieldPath: tags
+          toFieldPath: spec.forProvider.tags
+```
+
+The [EnvironmentConfigs]({{<ref "./environment-configs" >}}) page has 
+more information on EnvironmentConfigs options and usage.
+
+### Composite functions
+
+Composite functions (`XFNs`) are containers executing any code you define.
+Composite functions can read and write to any resource in the Composition
+they're attached to. This allows composite functions to preform complex
+operations to patch fields, determine if a resource is ready for use or notify
+an external system about the details of resource. 
+
+{{<hint "important" >}}
+Composite functions are an alpha feature. Alpha features aren't enabled by
+default. 
+{{< /hint >}}
+
+To attach a composite function to a Composition define a 
+{{<hover label="xfn" line="7">}}function{{</hover>}} inside the Composition
+{{<hover label="xfn" line="4">}}spec{{</hover>}}.
+
+Provide a {{<hover label="xfn" line="8">}}name{{</hover>}} for the function.   
+The {{<hover label="xfn" line="9">}}type{{</hover>}} is always `Container`.  
+The {{<hover label="xfn" line="11">}}container.image{{</hover>}} is the
+location of the composite function container image.
+
+```yaml {label="xfn",copy-lines="none"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for Brevity
+spec:
+  resources:
+  # Removed for Brevity
+  functions:
+  - name: rds-instance-password
+    type: Container
+    container:
+      image: xpkg.upbound.io/provider-aws-xfns/random-rds-password:v0.1.0
+```
+
+Read the [composition functions]({{<ref "./composition-functions">}}) page for
+details on building and using composite functions.
+
+
+
+### Storing connection details
+
+Some managed resources generate unique details like usernames, passwords, 
+IP addresses, ports or other connection details. 
+
+{{<hint "note" >}}
+The managed resource defines the generated details.   
+Read the
+[Managed Resources]({{<ref "./managed-resources#writeconnectionsecrettoref" >}})
+page for more information about connection secrets.
+{{</hint >}}
+
+The Composition defines where to save the details generated by any resources.
+#### Save connection details to a namespace
+By default Crossplane saves connection details in the `default` namespace. A
+Composition can save connection details to a specific namespace with
+`writeConnectionSecretsToNamespace`. 
```

Comment:
I believe what's this is configuring is a little more niche.

`writeConnectionSecretsToNamespace` specifies what namespace an XR (only - not its composed MRs) should write its connection details to if that XR is automatically created by Crossplane because someone created a claim.

If you create an XR directly/explicitly you must set `spec.writeConnectionDetailsTo.namespace` on the XR. If you're writing a Composition you must specify `spec.writeConnectionDetailTo.namespace` for any composed MR.

It's only when you create a claim that we don't really know what namespace the XR should use to store its secrets, so we derive it from the Composition.